core/embed/lib/gfx_draw.c

```diff
@@ -241,6 +241,53 @@ void gfx_draw_text(gfx_offset_t pos, const char* text, size_t maxlen,
   }
 }
 
+#ifdef TREZOR_PRODTEST
+
+#include "qr-code-generator/qrcodegen.h"
+#define QR_MAX_VERSION 9
+
+void gfx_draw_qrcode(int x, int y, uint8_t scale, const char* data) {
+  if (scale < 1 || scale > 10) return;
+
+  uint8_t codedata[qrcodegen_BUFFER_LEN_FOR_VERSION(QR_MAX_VERSION)] = {0};
+  uint8_t tempdata[qrcodegen_BUFFER_LEN_FOR_VERSION(QR_MAX_VERSION)] = {0};
+
+  int side = 0;
+  if (qrcodegen_encodeText(data, tempdata, codedata, qrcodegen_Ecc_MEDIUM,
+                           qrcodegen_VERSION_MIN, QR_MAX_VERSION,
+                           qrcodegen_Mask_AUTO, true)) {
+    side = qrcodegen_getSize(codedata);
+  }
+
+  // Calculate border size (1 extra modules around the QR code)
+  int border_side = ((side + 2) * scale);
+
+  // Calculate border left-top corner
+  x -= border_side / 2;
+  y -= border_side / 2;
+
+  // Draw border around the QR code
+  gfx_rect_t border_rect = gfx_rect_wh(x, y, border_side, border_side);
+  gfx_draw_bar(border_rect, COLOR_WHITE);
```

Comment:
Taking COLOR_WHITE from colors.h - where it is only defined for 16bit color depths. We should make this work for 32bit colors too.

Reply:
Yes that problem, but how about leaving the `COLOR_WHITE/BLACK` issue until we remove the original display driver? Once `dma2d.h/.c` is gone, `colors.h/.c` will go with it. Then we can move those constants to `gfx_color.h` and have there both 16-bit and 32-bit versions.

Reply:
Ok we can postpone this.

In case we need to support 32bit colors here sooner than we can drop old rendering we will revisit this.